src/aurelia-slickgrid/models/column.interface.ts

```diff
@@ -96,7 +106,7 @@ export interface Column<T = any> {
    * NOTE: a field with dot notation (.) will be considered a complex object.
    * For example: { id: 'Users', field: 'user.firstName' }
    */
-  field: string;
+  field: Join<PathsToStringProps<T>, '.'>;
```

Comment:
quite a feat, can that be `string | Join<PathsToStringProps<T>, '.'>` or it's unnecessary? What does it show in the intellisense? I wouldn't mind seeing print screen of that in the PR

Reply:
Not next to a PC currently but if you Union with string this one wins since it includes all the Options from the other. In intellisense it shows the available dot.notations for T.

This one would certainly need more testing which is why i've created the draft so that you can fire your expectations at it. E.g can also a whole complex object be the field or only primitives? Not sure If thats currently handled